hbase-common/src/test/java/org/apache/hadoop/hbase/HBaseClassTestRule.java

```diff
@@ -59,13 +59,20 @@ private HBaseClassTestRule(Class<?> clazz, Timeout timeout) {
 
   private static long getTimeoutInSeconds(Class<?> clazz) {
     Category[] categories = clazz.getAnnotationsByType(Category.class);
-    for (Class<?> c : categories[0].value()) {
-      if (c == SmallTests.class || c == MediumTests.class || c == LargeTests.class) {
-        // All tests have a 13 minutes timeout.
-        return TimeUnit.MINUTES.toSeconds(13);
-      }
-      if (c == IntegrationTests.class) {
-        return TimeUnit.MINUTES.toSeconds(Long.MAX_VALUE);
+    // This should never happen
+    if (categories.length > 1) {
+      throw new IllegalArgumentException("Code-bug: unsure how to handle more than one Category");
+    }
+    // Fail gracefully if there is no Category defined
+    if (categories.length == 1) {
+      for (Class<?> c : categories[0].value()) {
+        if (c == SmallTests.class || c == MediumTests.class || c == LargeTests.class) {
```

Comment:
nit: good to create immutable final set/list of all 3 values and then check with `if set.contains(c)`?

Reply:
I like it.